counterparty-server kickstart reparse error: Incorrect ledger_hash

```
[INFO] Block (re‐parse): 334000
[ERROR] Unhandled Exception
Traceback (most recent call last):
  File "/bin/counterparty-server", line 12, in <module>
    server.main()
  File "/usr/lib/python3.4/site-packages/counterpartycli/server.py", line 185, in main
    server.kickstart(db, bitcoind_dir=args.bitcoind_dir)
  File "/usr/lib/python3.4/site-packages/counterpartylib/server.py", line 345, in kickstart
    blocks.kickstart(db, bitcoind_dir=bitcoind_dir)
  File "/usr/lib/python3.4/site-packages/counterpartylib/lib/blocks.py", line 832, in kickstart
    reparse(db)
  File "/usr/lib/python3.4/site-packages/counterpartylib/lib/blocks.py", line 673, in reparse
    previous_ledger_hash, previous_txlist_hash)
  File "/usr/lib/python3.4/site-packages/counterpartylib/lib/blocks.py", line 159, in parse_block
    new_ledger_hash = check.consensus_hash(db, 'ledger_hash', previous_ledger_hash, util.BLOCK_LEDGER)
  File "/usr/lib/python3.4/site-packages/counterpartylib/lib/check.py", line 80, in consensus_hash
    raise ConsensusError('Incorrect {} for block {}.'.format(field, block_index))
counterpartylib.lib.check.ConsensusError: Incorrect ledger_hash for block 334000.
[1]+  Aborted                 (core dumped) bitcoind
```

<bountysource-plugin>

---
Want to back this issue? **[Place a bounty on it!](https://www.bountysource.com/issues/7858800-counterparty-server-kickstart-reparse-error-incorrect-ledger_hash?utm_campaign=plugin&utm_content=tracker%2F494307&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F494307&utm_medium=issues&utm_source=github).
</bountysource-plugin>

It wasn't, but the issue remains with the newer commit:

```
[INFO] Block (re‐parse): 334000
[ERROR] Unhandled Exception
Traceback (most recent call last):
  File "/bin/counterparty-server", line 12, in <module>
    server.main()
  File "/usr/lib/python3.4/site-packages/counterpartycli/server.py", line 182, in main
    server.kickstart(db, bitcoind_dir=args.bitcoind_dir)
  File "/usr/lib/python3.4/site-packages/counterpartylib/server.py", line 341, in kickstart
    blocks.kickstart(db, bitcoind_dir=bitcoind_dir)
  File "/usr/lib/python3.4/site-packages/counterpartylib/lib/blocks.py", line 833, in kickstart
    reparse(db)
  File "/usr/lib/python3.4/site-packages/counterpartylib/lib/blocks.py", line 674, in reparse
    previous_ledger_hash, previous_txlist_hash)
  File "/usr/lib/python3.4/site-packages/counterpartylib/lib/blocks.py", line 159, in parse_block
    new_ledger_hash = check.consensus_hash(db, 'ledger_hash', previous_ledger_hash, util.BLOCK_LEDGER)
  File "/usr/lib/python3.4/site-packages/counterpartylib/lib/check.py", line 80, in consensus_hash
    raise ConsensusError('Incorrect {} for block {}.'.format(field, block_index))
counterpartylib.lib.check.ConsensusError: Incorrect ledger_hash for block 334000.
```

--- a/counterpartylib/lib/api.py
+++ b/counterpartylib/lib/api.py
@@ -592,7 +592,11 @@ class APIServer(threading.Thread):
                 caught_up = True
 
             try:
-                last_block = util.CURRENT_BLOCK_INDEX
+                cursor = db.cursor()
+                blocks = list(cursor.execute('''SELECT * FROM blocks WHERE block_index = ?''', (util.CURRENT_BLOCK_INDEX, )))
+                assert len(blocks) == 1
+                last_block = blocks[0]
+                cursor.close()
             except:
                 last_block = None
 

--- a/counterpartylib/lib/blocks.py
+++ b/counterpartylib/lib/blocks.py
@@ -156,8 +156,8 @@ def parse_block(db, block_index, block_time, previous_ledger_hash=None,
     cursor.close()
 
     # Consensus hashes.
-    new_ledger_hash = check.consensus_hash(db, 'ledger_hash', previous_ledger_hash, util.BLOCK_LEDGER)
     new_txlist_hash = check.consensus_hash(db, 'txlist_hash', previous_txlist_hash, txlist)
+    new_ledger_hash = check.consensus_hash(db, 'ledger_hash', previous_ledger_hash, util.BLOCK_LEDGER)
 
     return new_ledger_hash, new_txlist_hash
 

--- a/counterpartylib/lib/blocks.py
+++ b/counterpartylib/lib/blocks.py
@@ -357,7 +357,7 @@ def get_tx_info(tx_hex, block_parser=None, block_index=None):
     if not block_index:
         block_index = util.CURRENT_BLOCK_INDEX
     try:
-        if util.enabled('multisig_addresses'):   # Protocol change.
+        if util.enabled('multisig_addresses', block_index=block_index):   # Protocol change.
             tx_info = get_tx_info2(tx_hex, block_parser=block_parser)
         else:
             tx_info = get_tx_info1(tx_hex, block_index, block_parser=block_parser)

--- a/counterpartylib/lib/blocks.py
+++ b/counterpartylib/lib/blocks.py
@@ -829,8 +829,8 @@ def kickstart(db, bitcoind_dir):
         cursor.execute('''UPDATE transactions SET tx_index = tx_index + ?''', (tx_index,))
         logger.info('Reordered transactions in {:.3f}s.'.format(time.time() - start_time))
 
-        # Parse all transactions in database.
-        reparse(db)
+    # Parse all transactions in database.
+    reparse(db)
 
     cursor.close()
     logger.info('Total duration: {:.3f}s'.format(time.time() - start_time_total))

--- a/counterpartylib/lib/util.py
+++ b/counterpartylib/lib/util.py
@@ -572,14 +572,17 @@ def unhexlify(hex_string):
     return binascii.unhexlify(bytes(hex_string, 'utf-8'))
 
 ### Protocol Changes ###
-def enabled (change_name):
+def enabled(change_name, block_index=None):
     """Return True if protocol change is enabled."""
     enable_block_index = PROTOCOL_CHANGES[change_name]['block_index']
+
+    if not block_index:
+        block_index = CURRENT_BLOCK_INDEX
 
     if config.TESTNET: 
         return True     # Protocol changes are always retroactive on testnet.
     else:
-        if CURRENT_BLOCK_INDEX >= enable_block_index:
+        if block_index >= enable_block_index:
             return True
         else:
             return False

